torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -2,11 +2,65 @@
 #include <torch/csrc/jit/backends/backend_detail.h>
 #include <torch/csrc/jit/backends/backend_resolver.h>
 #include <torch/csrc/jit/frontend/code_template.h>
+#include <torch/csrc/jit/python/module_python.h>
 #include <torch/csrc/jit/python/pybind_utils.h>
 
 namespace torch {
 namespace jit {
 
+// Selectively lower \p cloned_mod to a backend. \p to_backend
+// is called to lower modules. \p modules_to_lower contains
+// qualified names of submodules of \p cloned_mod that should be lowered.
+void to_backend_selective_impl(
+    Module& cloned_mod,
+    py::function to_backend,
+    const std::vector<std::string>& modules_to_lower) {
+  // For each module that should be lowered:
+  for (const auto& module_to_lower : modules_to_lower) {
+    // Use QualifiedName to parse the qualified module names.
+    c10::QualifiedName qual_module_name(module_to_lower);
+    auto& atoms = qual_module_name.atoms();
+
+    // Search through the module hierarchy using the atoms of
+    // qual_module_name until current points to the module to
+    // be lowered. Make parent point to its parent for type
+    // modification purposes.
+    Module parent = cloned_mod;
+    Module current = cloned_mod;
+    for (size_t i = 0, e = atoms.size(); i < e; ++i) {
+      IValue submodule = current.attr(atoms[i]);
+      if (submodule.isModule()) {
+        // If this is the last atom, store a reference to the
+        // parent of the module that is about to be lowered.
+        if (i == e - 1) {
+          parent = current;
+        }
+        current = submodule.toModule();
+      } else {
+        std::stringstream err;
+        err << "Attribute named " << atoms[i] << " is not a Module";
+        throw std::runtime_error(err.str());
+      }
+    }
+
+    // Call to_backend on the module that needs to be lowered. It needs to be
+    // wrapped before doing so because _to_jit_backend accepts wrapped modules.
+    // The result needs to be unwrapped in order to access its type below.
+    auto lowered_submodule =
+        py::cast<Module>(to_backend(py::module::import("torch.jit._recursive")
+                                        .attr("wrap_cpp_module")(current))
+                             .attr("_c"));
+
+    // Adjust the parent's type so that the type of the submodule matches
+    // the type of lowered_submodule.
+    auto parent_jit_type = parent.type();
+    parent_jit_type->unsafeRemoveAttribute(qual_module_name.name());
```

Comment:
This changes the ordering of the attributes in `parent_jit_type`, which might be surprising (since our slot representations are ordered).

Reply:
Ah that's true, so should I remove all of them and add them back? Or since we're doing unsafe type mods anyway, I'm going to add an `unsafeSwapAttribute` so we can override the `ClassAttribute` struct for that slot instead of adding and removing.

Reply:
I think if you do the `type_remapper` thing I suggested below, you don't need to add anything new, since the remapping will happen while the type is cloned (and thus ordering will be preserved)